lib/vector-config-macros/src/configurable_component.rs

```diff
@@ -1,64 +1,231 @@
-use darling::{util::Flag, FromMeta};
+use darling::{util::path_to_string, Error, FromMeta};
 use proc_macro::TokenStream;
 use quote::quote;
 use syn::{
-    parse_macro_input, parse_quote_spanned, punctuated::Punctuated, token::Comma, AttributeArgs,
-    DeriveInput, Path,
+    parse_macro_input, parse_quote, parse_quote_spanned, punctuated::Punctuated, token::Comma,
+    AttributeArgs, DeriveInput, Ident, Lit, Meta, NestedMeta, Path,
 };
 
+#[derive(Copy, Clone)]
+struct AttributeIdent(&'static str);
+
+const ENRICHMENT_TABLE: AttributeIdent = AttributeIdent("enrichment_table");
+const PROVIDER: AttributeIdent = AttributeIdent("provider");
+const SINK: AttributeIdent = AttributeIdent("sink");
+const SOURCE: AttributeIdent = AttributeIdent("source");
+const TRANSFORM: AttributeIdent = AttributeIdent("transform");
+const NO_SER: AttributeIdent = AttributeIdent("no_ser");
+const NO_DESER: AttributeIdent = AttributeIdent("no_deser");
```

Comment:
I'm not seeing these two attributes actually used anywhere, other than in the macro code, of course. Leftovers, or did I miss something?

Reply:
They're used primarily in the sink helper types (compression, concurrency) and the encoding/transformer types.

Reply:
Oh, curse GitHub, not all patches are expanded so text search missed it.